### ISSUE TITLE
confluent_api_key : how to retrieve the key itself

### ISSUE THREAD

**Issue by neuweiler** [confluentinc/terraform-provider-confluent#31]
Via confluent_api_key.secret it is possible to access the key's secret but how to access the key (name, e.g. "AB6CD2EFGHIJK1LM") itself? confluent_api_key.key as suggested in other tickets doesn't work. Please also add this to the documentation.

BTW: It would also be a good idea to mention in the documentation that the secret itself can be read out and stored e.g. in a key-vault if human access is required.

**Comment by linouk23** [comment]
👋 thanks for creating an issue!

> Via confluent_api_key.secret it is possible to access the key's secret but how to access the key (name, e.g. "AB6CD2EFGHIJK1LM") itself? confluent_api_key.key as suggested in other tickets doesn't work. Please also add this to the documentation.

That's kinda similar to #30 (we'll add it to our docs in our next release!), you basically need to use `.id` attribute like this:
```
resource "confluent_kafka_topic" "orders" {
...
  credentials {
    key    = confluent_api_key.app-manager-kafka-api-key.id
    secret = confluent_api_key.app-manager-kafka-api-key.secret
  }
}
```

> BTW: It would also be a good idea to mention in the documentation that the secret itself can be read out and stored e.g. in a key-vault if human access is required.

sure, could you expand on that a little bit?

**Comment by linouk23** [comment]
Added `id` to the list of exported attributes of `confluent_api_key` [resource](https://registry.terraform.io/providers/confluentinc/confluent/latest/docs/resources/confluent_api_key#attributes-reference): 
![image](https://user-images.githubusercontent.com/5459870/168343047-c3d4b564-baaa-4cc9-9e40-92c5ab8ac071.png)
in the latest `0.8.1` release.

@neuweiler could you share more details about:

> BTW: It would also be a good idea to mention in the documentation that the secret itself can be read out and stored e.g. in a key-vault if human access is required

**Comment by Marcus-James-Adams** [comment]
@linouk23  something I have noticed - on a lot of your docs is that you are putting  (required String) or (optional) on attributes referenced.  from the terraform docs naming convention you don't put that on the description of any listed in the reference section as they are all references

**Comment by linouk23** [comment]
@Marcus-James-Adams that sounds interesting, could you expand on that? I didn't quite understand it:

> from the terraform docs naming convention you don't put that on the description of any listed in the reference section as they are all references

**Comment by linouk23** [comment]
@neuweiler we're happy to let you know we've just released `0.9.0` [version](https://github.com/confluentinc/terraform-provider-confluent/blob/master/CHANGELOG.md#090-may-25-2022) of TF Provider that fixes this issue in our docs.
